Modules/Quotas/src/Quotas.Domain/Aggregates/Identities/Identity.cs

```diff
@@ -129,4 +129,18 @@ private IEnumerable<Quota> GetAppliedQuotasForMetric(MetricKey metric)
         var appliedQuotas = allQuotasOfMetric.Where(q => q.Weight == highestWeight).ToArray();
         return appliedQuotas;
     }
+
+    public async Task ChangeTierAsync(Tier newTier, MetricCalculatorFactory metricCalculatorFactory, CancellationToken cancellationToken)
+    {
+        _tierQuotas.RemoveAll(_ => true);
+        _metricStatuses.RemoveAll(_ => true);
+
+        TierId = newTier.Id;
+        foreach (var tierQuotaDefinition in newTier.Quotas)
+        {
+            AssignTierQuotaFromDefinition(tierQuotaDefinition);
+        }
+        var metricKeys = newTier.Quotas.Select(q => q.MetricKey);
```

Comment:
What if there are multiple quotas on the same metric?
=> the same MetricStatus is updated multiple times
=> unnecessary work